src/main/java/frc/team88/swerve/configuration/Configuration.java

```diff
@@ -260,6 +262,27 @@ private NavX instantiateNavX(Config gyroConfig) {
     }
   }
 
+  /**
+   * Instatiates a Pigeon object.
+   *
+   * @param gyroConfig The config for the gyro.
+   * @return The Pigeon object.
+   * @throws SwerveConfigException If the user provided config is incorrect.
+   */
+  private Pigeon instantiatePigeon(Config gyroConfig) {
+    String portType = configCheckAndGet(gyroConfig, "port-type", String.class);
+    switch (portType) {
+      case "TalonSRX":
+        TalonSRX controller = new TalonSRX(configCheckAndGet(gyroConfig, "id", Integer.class));
+        return new Pigeon(controller);
+      case "CAN":
+        return new Pigeon(configCheckAndGet(gyroConfig, "id", Integer.class));
```

Comment:
I'd recommend doing an extra check to make sure the CAN ID is between 0 and 64. Same with the CAN option.

```suggestion
    String portType = configCheckAndGet(gyroConfig, "port-type", String.class);
    int id = configCheckAndGet(instanceConfig, "id", Integer.class);
    if (id< 0 || id >= 64) {
      throw new InvalidConfigValueException(
              String.format("id %d is not in range [0, 63]", id));
    }

    switch (portType) {
      case "TalonSRX":
        TalonSRX controller = new TalonSRX(id);
        return new Pigeon(controller);
      case "CAN":
        return new Pigeon(id);
```